testing/camunda-process-test-java/src/test/java/io/camunda/process/test/api/ElementAssertTest.java

```diff
@@ -1378,6 +1378,146 @@ void shouldWaitUntilElementsAreEnded() {
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("A", "C");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("B", "C");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("A");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("B");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("C");
+    }
+
+    @Test
+    void shouldPassIfElementsAreCompletedInOrderForTheSameInstance() {
+      // given
+      when(camundaDataSource.findFlowNodeInstances(any()))
+          .thenReturn(
+              Arrays.asList(
+                  newCompletedFlowNodeInstance("A"),
+                  newCompletedFlowNodeInstance("A"),
+                  newCompletedFlowNodeInstance("A")));
+
+      // when
+      when(processInstanceEvent.getProcessInstanceKey()).thenReturn(PROCESS_INSTANCE_KEY);
+
+      // then
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("A", "A", "A");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("A", "A");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("A");
+    }
+
+    @Test
+    void shouldPassIfElementsAreCompletedInOrderMultipleTimes() {
+      // given
+      when(camundaDataSource.findFlowNodeInstances(any()))
+          .thenReturn(
+              Arrays.asList(
+                  newCompletedFlowNodeInstance("A"),
+                  newCompletedFlowNodeInstance("A"),
+                  newCompletedFlowNodeInstance("B"),
+                  newCompletedFlowNodeInstance("A")));
+
+      // when
+      when(processInstanceEvent.getProcessInstanceKey()).thenReturn(PROCESS_INSTANCE_KEY);
+
+      // then
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("A", "B", "A");
+      CamundaAssert.assertThat(processInstanceEvent).hasCompletedElementsInOrder("B", "A");
+    }
+
+    @Test
+    void shouldFailIfOneElementIsNotCompleted() {
+      // given
+      when(camundaDataSource.findFlowNodeInstances(any()))
+          .thenReturn(
+              Arrays.asList(
+                  newCompletedFlowNodeInstance("A"),
+                  newCompletedFlowNodeInstance("B"),
+                  newActiveFlowNodeInstance("A")));
+
+      // when
+      when(processInstanceEvent.getProcessInstanceKey()).thenReturn(PROCESS_INSTANCE_KEY);
+
+      // then
+      Assertions.assertThatThrownBy(
+              () ->
+                  CamundaAssert.assertThat(processInstanceEvent)
+                      .hasCompletedElementsInOrder("A", "B", "A"))
+          .hasMessage(
+              "Process instance [key: %d] should have completed elements ['A', 'B', 'A'] in order, but only the following elements were encountered:\n"
+                  + "\t- 'A': completed\n"
+                  + "\t- 'B': completed\n"
+                  + "\t- 'A': active",
```

Comment:
:wrench: We should shrink the failure message and list only the completed elements. 

If the test fails, the result collector prints all active element instances of the process instance. Having the same active elements in the failure message would be redundant. 

```suggestion
              "Process instance [key: %d] should have completed elements ['A', 'B', 'A'] in order, but only the following elements were completed:\n"
                  + "\t- 'A': completed\n"
                  + "\t- 'B': completed",
```